geoengine/datasets.py

```diff
@@ -4,37 +4,41 @@
 
 from __future__ import annotations
 from abc import abstractmethod
-from datetime import datetime
-from typing import Dict, List, NamedTuple, Optional, Tuple, Union, Generic, TypeVar
-
+from typing import List, NamedTuple, Optional, cast
 from enum import Enum
 from uuid import UUID
-
 import json
-from typing_extensions import Literal, TypedDict
+from typing_extensions import Literal
 from attr import dataclass
 import numpy as np
 import geopandas as gpd
 import requests as req
-
+from geoengine import api
 from geoengine.error import GeoEngineException, InputException
 from geoengine.auth import get_session
-from geoengine.types import GdalDatasetParameters, RasterResultDescriptor, TimeStep, TimeStepGranularity, VectorDataType
+from geoengine.types import Provenance, RasterSymbology, TimeStep, \
+    TimeStepGranularity, VectorDataType
 
 
-_OrgSourceDurationDictT = TypeVar('_OrgSourceDurationDictT', str, Union[str, int, TimeStepGranularity])
+class UnixTimeStampType(Enum):
+    '''A unix time stamp type'''
+    EPOCHSECONDS = 'epochSeconds'
+    EPOCHMILLISECONDS = 'epochMilliseconds'
+
+    def to_api_enum(self) -> api.UnixTimeStampType:
+        return api.UnixTimeStampType(self.value)
 
 
 class OgrSourceTimeFormat:
     '''Base class for OGR time formats'''
 
     @abstractmethod
-    def to_dict(self) -> Dict[str, str]:
+    def to_api_dict(self) -> api.OgrSourceTimeFormat:
         pass
 
     @classmethod
-    def seconds(cls) -> SecondsOgrSourceTimeFormat:
-        return SecondsOgrSourceTimeFormat()
+    def seconds(cls, timestamp_type: UnixTimeStampType) -> UnixTimeStampOgrSourceTimeFormat:
```

Comment:
What is this? We have milliseconds all the time, right?

Reply:
i have no idea. I think i just moved it. This is also in the backend: https://github.com/geo-engine/geoengine/blob/660cfff25779acf7abe9bd61e6b9700b49c0035a/operators/src/source/ogr_source/mod.rs#L207